src/app/blog/BlogClient.tsx

```diff
@@ -52,42 +77,72 @@ export function BlogClient({ posts }: { posts: BlogPostData[] }) {
         onChange={handleSearch}
       />
-      <ul>
-        {filteredPosts.slice(0, visibleCount).map((post) => (
-          <li key={post.slug}>
-            {post.image.url && (
-              <Image
-                src={post.image.url}
-                alt={post.image.alt}
-                width={282}
-                height={141}
-                className="object-cover"
-              />
-            )}
-            <Heading tag="h3" variant="lg">
-              {post.title}
-            </Heading>
-            <p>{post.description}</p>
-            {post.publishedOn && (
-              <span className="block">
-                {formatDate(post.publishedOn, 'blog')}
-              </span>
-            )}
-            <TextLink href={`${PATHS.BLOG.path}/${post.slug}`}>
-              Read More
-            </TextLink>
-          </li>
-        ))}
-      </ul>
-      {hasMorePosts && (
-        <Button aria-label="Load more posts" onClick={handleLoadMore}>
-          Load More
-        </Button>
+
+      <br />
+      <br />
+
+      <CardLayout type="blogPost">
+        {filteredPosts.map((post, i) => {
+          const { image, title, description, slug, publishedOn } = post
+
+          return (
+            <li
+              key={post.slug}
+              className={clsx(
+                'h-[400px] overflow-clip rounded-md border border-brand-600 p-4',
+                // Only show 20 posts at a time, the rest are hidden with sr-only
+                i >= (currentPage - 1) * POSTS_PER_LOAD &&
+                  i < currentPage * POSTS_PER_LOAD
+                  ? 'block'
+                  : 'sr-only',
+              )}
+            >
+              {image.url && (
+                <Image
+                  src={image.url}
+                  alt={image.alt}
+                  width={282}
+                  height={141}
+                  className="object-cover"
+                />
+              )}
+              <Heading tag="h3" variant="lg">
+                {title}
+              </Heading>
+              <p>{description}</p>
+              {publishedOn && (
+                <span className="block">{formatDate(publishedOn, 'blog')}</span>
+              )}
+              <TextLink href={`${PATHS.BLOG.path}/${slug}`}>Read More</TextLink>
+            </li>
+          )
+        })}
+      </CardLayout>
+
+      {!filteredPosts.length && (
+        <p className="mt-8 rounded-md border border-brand-600 p-4">
+          No results found for your search, try changing your search query.
+        </p>
       )}
+
+      <div className="mx-auto mt-8 max-w-2xl">
+        <ClientPagination
+          currentPage={currentPage}
+          total={filteredPosts.length}
+          size={POSTS_PER_LOAD}
+          setCurrentPage={setCurrentPage}
+        />
+      </div>
     </>
   )
+
+  function handleSearch(event: React.ChangeEvent<HTMLInputElement>) {
+    setCurrentPage(1)
+    setSearchQuery(event.target.value)
+  }
```

Comment:
Can we please keep the function declarations at the top?

Reply:
No problem.

Reply:
I know. I read you blog posts. ❤️